src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/markup_rope_inputs.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "markup_rope_inputs.hpp"
+
+#include <unordered_set>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/rotary_positional_embeddings.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+#include "transformations/utils/gen_pattern.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::intel_cpu::MarkUpRopeInputs::MarkUpRopeInputs() {
+    MATCHER_SCOPE(MarkUpRopeInputs);
+    using namespace ov::pass::pattern;
+    using namespace ov::gen_pattern;
+    auto cos_tab = any_input();
+    auto sin_tab = any_input();
+    auto rope = makePattern<ov::op::internal::RoPE>({any_input(), cos_tab, sin_tab});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto cos_input_node = pattern_map.at(cos_tab).get_node_shared_ptr();
+        auto sin_input_node = pattern_map.at(sin_tab).get_node_shared_ptr();
+        auto bfs_markup = [&](std::shared_ptr<ov::Node>& input) {
```

Comment:
Please check if we can reuse `visit_path_impl` from [transformations utils](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/utils/utils.cpp)

Reply:
It sounds reasonable, let me try it.

Reply:
The `visit_path_impl` is not a public API, should we export it as a public API ?

Reply:
@v-Golubev do you think we should export the `visit_path_impl` as a public API ? Otherwise, other component cannot call it directly.

Reply:
I think it is reasonable, we can rename it to `visit_path` and export.
@itikhono don't you mind if we do that?

Reply:
@praasz if you are against this change please let us know as well

Reply:
Add this function to `src/common/transformations/include/transformations/utils/utils.hpp` will make it part Transformations API (it is dev API no public API) it should be fine as it can be used by other components in OV.

When move rename function and doxy comment.

Another option for new name `visit_node_path_if_not`. It should be some name which describes function that visit nodes from root node and apply function except nodes in predicate.

Also check the 2nd parameter `visited` in current utils is working as output parameter but it looks like is not used outside functions. Maybe new util function in dev API should not have it.